src/features.jl

```diff
@@ -0,0 +1,69 @@
+_flattenimage(image::Matrix) = vec(image)
+
+function _flattenimage{T}(images::Array{T,3})
+    ncols, nrows, nimages = size(images)
+    reshape(images, (nrows * ncols, nimages))
+end
+
+
+feature_doc(train_or_test) =
+"""
+    $(train_or_test)features([indices])
+
+Returns the MNIST **$(train_or_test)set** features denoted by `indices`.
+
+- if `indices` is an `Integer`, the single image is returned as
+flat `Vector{Float64}` feature vector.
+
+- if `indices` is a `AbstractVector`, the images are returned as
+a design matrix `Matrix{Float64}` in feature major layout, which
```

Comment:
Realted: What do you think about "3D matrix", "3D array", or "rank 3 tensor" instead of "3D tensor". (i used this term in the docstrings for images) 

While I feel that I _hear_ the term "3D tensor" constantly, I fail to find it used in literature and searching for it on the web suggests that this term maybe ill used in this case. I suspect there are some subtleties involved in the differences even between matrices and rank 2 tensors that are a little beyond my current knowledge. 

The safest bet in terms of correctness may be to just call it "3D array"